### PR TITLE
refactor: remove dicts in determine_node_settings

### LINKED ISSUE #173: Use Particle and spin projection within STM._determine_node_settings
Similar to the other Issues in the Epic, use the `Particle` class more in the implementation and use the adapter to convert it to a nested dict structure to keep the interface stable.

### PR DESCRIPTION
**Pull request by redeboer**
Closes #173

### CODE REVIEW

File: expertsystem/topology.py [hunk 1]
```diff
@@ -320,9 +320,7 @@ def __init__(
 
     def __repr__(self) -> str:
         return (
-            f"{self.__class__.__name__}()"
-            f"\n    nodes: {self.nodes}"
-            f"\n    edges: {self.edges}"
+            f"{self.__class__.__name__}(nodes={self.nodes}, edges={self.edges})"
             f"\n    node props: {self.node_props}"
             f"\n    node props: {self.edge_props}"
```
Comment (spflueger) [inline expertsystem/topology.py:325]: should be `edge props:`

File: expertsystem/ui/__init__.py [hunk 2]
```diff
@@ -237,20 +239,33 @@ def _create_seed_graphs(
...
+                            edge_id
+                        ] = particle_with_spin_projection_to_dict(  # type: ignore
+                            edge_property
+                        )
+
+        if isinstance(instance, list) and len(instance) > 0:
+            if isinstance(instance[0], StateTransitionGraph):
+                convert_edges_in_graph(instance)  # type: ignore
+            elif isinstance(instance[0], tuple):
+                convert_edges_in_graph([g for g, _ in instance])  # type: ignore
```
Comment (spflueger) [inline expertsystem/ui/__init__.py:265]: does this work? since a tuple is immutable it should be not possible to alter it in place

**Review by spflueger — CHANGES_REQUESTED** [review]
Nice. Just two small issues

File: expertsystem/topology.py (hunk 1 above)
Comment (redeboer) [reply expertsystem/topology.py]: 131b22cf
Actually this `__repr__` is still nonsensical and it's also still quite obscure when printing e.g. a `list` of `StateTransitionGraphs` (namely the solutions the STM finds), but a for a better `__repr__` we may have to reconsider the constructor of this class (perhaps even implement a `__new__`).

File: expertsystem/ui/__init__.py (hunk 2 above)
Comment (redeboer) [reply expertsystem/ui/__init__.py:265]: Yes, it's a `tuple` of two, so you unpack the first into `g` and the second into `_`

File: expertsystem/topology.py (hunk 1 above)
Comment (spflueger) [reply expertsystem/topology.py]: Thats true...

**Pull request merged**